core/src/test/java/io/grpc/internal/BinaryLogProviderTest.java

```diff
@@ -310,6 +321,53 @@ public void sendMessage(RespT message) {
     return methodDef.getServerCallHandler().startCall(serverCall, new Metadata());
   }
 
+  @Test
+  public void serverCallIdSetter() {
+    ServerStreamTracer tracer = BinaryLogProvider.SERVER_CALLID_SETTER_FACTORY
+        .newServerStreamTracer(
+            "service/method", new Metadata());
+    MockableSpan mockableSpan = MockableSpan.generateRandomSpan(new Random(0));
+    Context context = Context.current().withValue(CONTEXT_SPAN_KEY, mockableSpan);
+    Context filtered = tracer.filterContext(context);
+    CallId callId = BinaryLogProvider.SERVER_CALL_ID_CONTEXT_KEY.get(filtered);
+    assertTrue(Arrays.equals(mockableSpan.getContext().getTraceId().getBytes(), callId.id));
```

Comment:
`assertThat(mockableSpan.getContext().getTraceId().getBytes()).isEqualTo(callId.id)`

Reply:
done here and below